### PR TITLE
Make modules information in publish workflow configurable - master

### PR DESCRIPTION
**Pull request by PeterSuna**
lighty-rnc test run: https://github.com/PeterSuna/lighty-core/runs/3558544247?check_suite_focus=true
lighty-rcgnmi test run: https://github.com/PeterSuna/lighty-core/runs/3558642055?check_suite_focus=true

Signed-off-by: PeterSuna <Peter.Suna@pantheon.tech>

### CODE REVIEW

File: .github/workflows/publish-docker-helm.yml [hunk 1]
```diff
@@ -57,7 +57,7 @@ jobs:
           echo "DOCKER_IMAGE_NAME=$(echo $DOCKER_IMAGE_NAME)" >> $GITHUB_ENV
           echo "DOCKER_IMAGE_NAME_TAG=$(echo $DOCKER_IMAGE_NAME_TAG)" >> $GITHUB_ENV
       - name: Build docker image
-        run: mvn install -B -pl :${{ github.event.inputs.app-name }}-module,:${{ github.event.inputs.app-name }}-app,:${{ github.event.inputs.app-name }}-app-docker -P docker
+        run: mvn install -B -pl :${{ github.event.inputs.app-name }}-app-module,:${{ github.event.inputs.app-name }}-app,:${{ github.event.inputs.app-name }}-app-docker -P docker
```
Comment (Pyro2266) [inline .github/workflows/publish-docker-helm.yml:60]: Being dependent on format of artifact names is not a good idea. We don't follow this format strictly and it is not documented anywhere.
Even two `lighty-applications` we have use different format. Module in RNC is called just lighty-rnc-**module** while in RCgNMI it's called lighty-rcgnmi-**app-module**. So this postfix you added will probably not work for RNC.

**Review by Pyro2266 — CHANGES_REQUESTED** [review]

**Review by vainyksi — CHANGES_REQUESTED** [review]
extract artifact names & paths used in workflow to input parameters - that provides us possibility to generalize the other new workflows per lighty-app or automate execution later

**Review by vainyksi — CHANGES_REQUESTED** [review]

File: .github/workflows/publish-docker-helm.yml [hunk 2]
```diff
@@ -57,10 +72,10 @@ jobs:
           echo "DOCKER_IMAGE_NAME=$(echo $DOCKER_IMAGE_NAME)" >> $GITHUB_ENV
           echo "DOCKER_IMAGE_NAME_TAG=$(echo $DOCKER_IMAGE_NAME_TAG)" >> $GITHUB_ENV
       - name: Build docker image
-        run: mvn install -B -pl :${{ github.event.inputs.app-name }}-module,:${{ github.event.inputs.app-name }}-app,:${{ github.event.inputs.app-name }}-app-docker -P docker
+        run: mvn install -B -pl :${{ github.event.inputs.app-modules }} -P docker
```
[the remark this answers is not in the document]
Comment (PeterSuna) [reply .github/workflows/publish-docker-helm.yml]: done

File: .github/workflows/publish-docker-helm.yml [hunk 3]
```diff
@@ -6,6 +6,22 @@ on:
         description: Name of the application in /lighty-applications
         default: lighty-rnc
         required: true
+      app-modules:
+        description: Name of the modules with format ':{module-name},...' in /lighty-applications required for build
```
[the remark this answers is not in the document]
Comment (PeterSuna) [reply .github/workflows/publish-docker-helm.yml]: done

File: .github/workflows/publish-docker-helm.yml [hunk 4]
```diff
@@ -6,6 +6,22 @@ on:
...
+        default: :lighty-rnc-app,:lighty-rnc-app-docker,:lighty-rnc-module
```
[the remark this answers is not in the document]
Comment (PeterSuna) [reply .github/workflows/publish-docker-helm.yml]: done

File: .github/workflows/publish-docker-helm.yml [hunk 5]
```diff
@@ -6,6 +6,22 @@ on:
...
+        required: true
+      app-docker-pom-path:
+        description: Path to docker pom file in /lighty-applications
+        default: lighty-applications/lighty-rnc-app-aggregator/lighty-rnc-app-docker/pom.xml
```
[the remark this answers is not in the document]
Comment (PeterSuna) [reply .github/workflows/publish-docker-helm.yml]: done

File: .github/workflows/publish-docker-helm.yml [hunk 6]
```diff
@@ -6,6 +6,22 @@ on:
...
+        required: true
+      app-helm-values-path:
+        description: Path to Helm YAML values in /lighty-applications
+        default: lighty-applications/lighty-rnc-app-aggregator/lighty-rnc-app-helm/helm/lighty-rnc-app-helm/values.yaml
+        required: true
+      app-helm-folder-path:
+        description: Path to Helm folde in /lighty-applications
+        default: lighty-applications/lighty-rnc-app-aggregator/lighty-rnc-app-helm/helm/
```
[the remark this answers is not in the document]
Comment (PeterSuna) [reply .github/workflows/publish-docker-helm.yml]: done

File: .github/workflows/publish-docker-helm.yml [hunk 7]
```diff
@@ -6,6 +6,18 @@ on:
         description: Name of the application in /lighty-applications
         default: lighty-rnc
```
Comment (vainyksi) [inline .github/workflows/publish-docker-helm.yml:7]: defaults for `app-name` & also `image-name` bellow are also redundant now - make them fully required to enter by user without default values

File: .github/workflows/publish-docker-helm.yml [hunk 8]
```diff
@@ -6,6 +6,18 @@ on:
...
         required: true
+      app-modules:
+        description: Name of the modules with format ':{module-name},...' required for build
```
Comment (vainyksi) [inline .github/workflows/publish-docker-helm.yml:10]: "... require to build **docker image**"

**Review by vainyksi — CHANGES_REQUESTED** [review]

**Review by vainyksi — APPROVED** [review]

File: .github/workflows/publish-docker-helm.yml (hunk 7 above)
Comment (PeterSuna) [reply .github/workflows/publish-docker-helm.yml]: done

File: .github/workflows/publish-docker-helm.yml (hunk 8 above)
Comment (PeterSuna) [reply .github/workflows/publish-docker-helm.yml]: done

**Review by Pyro2266 — APPROVED** [review]

**Pull request merged**